javascript/javascript1/week4/index.js

```diff
@@ -0,0 +1,161 @@
+  }
+
+  // remove todo item from todolist array
+  else if (command.toLowerCase().includes("remove")) {
+    console.log(todolist);
+    // split the command and get rid of first one(Add),
+    tempArray.shift(tempArray[0]);
+    let todo = "";
+
+    // and last three(to my todo)
+    for (let i = 0; i < 3; i++) {
+      tempArray.pop(tempArray[tempArray.length - 1]);
+    }
+
+    // move or copy the new tempArray to todo
+    for (let a = 0; a < tempArray.length; a++) {
+      const element = tempArray[a];
+      todo += tempArray[a] + " ";
+    }
+
+    let removeItem = todolist.splice(todo, 1);
+    console.log(`Removed '${removeItem}' from your todo, ${userName}`);
+  }
+
+  // checking and showing current todolist status
+  else if (command.toLowerCase().includes("on my todo?")) {
+    console.log(
+      `You have ${todolist.length} todo(s).\n${userName}'s ToDo List = [${todolist}]`
+    );
+  }
+
+  // show today's date
+  else if (command.toLowerCase().includes("is it today?")) {
+    const monthNames = [
+      "January",
+      "February",
+      "March",
+      "April",
+      "May",
+      "June",
+      "July",
+      "August",
+      "September",
+      "October",
+      "November",
+      "December",
+    ];
+    const today = new Date();
+    var dd = String(today.getDate()).padStart(2, "0");
+    var yyyy = today.getFullYear();
+    console.log(`${dd}. of ${monthNames[today.getMonth()]} ${yyyy}`);
+  }
+
+  // make a simple calculator
+  else if (command.toLowerCase().includes("what is")) {
+    const match = command.match(/(\d+)\s*([\+\-\*\/])\s*(\d+)/);
+    if (match) {
+      if (match[2] === "+") {
+        result = parseInt(match[1]) + parseInt(match[3]);
+      }
+      if (match[2] === "-") {
+        result = parseInt(match[1]) - parseInt(match[3]);
+      }
+      if (match[2] === "*") {
+        result = parseInt(match[1]) * parseInt(match[3]);
+      }
+      if (match[2] === "/") {
+        result = parseInt(match[1]) / parseInt(match[3]);
+      }
+      console.log(result);
+    } else {
+      console.log("Sorry, please put right syntex");
+    }
+  }
+
+  // setting timer
+  else if (command.toLowerCase().includes("minutes")) {
+    const match = command.match(/\d+/); // detect the digit(should always take minutes)
+    if (match) {
+      console.log(`Timer set for ${match[0]} minutes`);
+
+      let timer = parseInt(match[0]) * 60 * 1000; //change the minute number into milliseconds
+
+      // const minutes = Math.floor(timer / 60);
+      // let seconds = timer % 60;
+      // seconds = seconds < 10 ? "0" + seconds : seconds;
+
+      // console.log(`${minutes} : ${seconds}`);
+
+      setTimeout(() => {
+        console.log(parseInt(match[0]) + " minute(s)" + "Timer done");
+      }, timer);
+    }
+  } else {
+    console.log("Sorry, I didn't understand you. Please try again");
+  }
+}
+
+console.log(getReply("Hello my name is Benjamin")); // "Nice to meet you benjamin"
```

Comment:
Consider sending the same command twice as the task says. Maybe respond with `Hey again, Benjamin!`